external links + unrecognized git

<!--
Ouch, sorry you’ve run into a bug.  Thank for taking the time to report it!

Please fill in as much of the template below as you’re able.

P.S. have you seen our support and contributing docs?
https://github.com/remarkjs/.github/blob/master/support.md
https://github.com/remarkjs/.github/blob/master/contributing.md
-->

### Subject of the issue

I think latest master has a problem with external links and a null git repository. My setup with v8.0.3 and the following config:

```js
'use strict';
module.exports = {
  'settings': {
    'bullet': '*',
    'fence': '`',
    'fences': true,
    'incrementListMarker': false,
  },
  'plugins': [
    'preset-lint-recommended',
    'preset-lint-consistent',
    // 'lint-blank-lines-1-0-2', buggy
    // [
    //   'lint-code',
    //   {
    //     js: 'remark-lint-code-eslint',
    //   },
    // ],
    // 'lint-no-empty-sections',
    'lint-heading-whitespace',
    'lint-no-url-trailing-slash',
    [
      'validate-links',
      {
        // https://github.com/remarkjs/remark-validate-links/issues/28
        'repository': 'unused/unused',
      },
    ],
  ],
};
```

gave me no issues. When I try latest master, I get

```
CONTRIBUTING.md
      18:53-18:84  warning  Link to unknown file: `https:/remark.js.org`                                                                  missing-file             remark-validate-links
     19:70-19:116  warning  Link to unknown file: `https:/github.com/OSSIndex/auditjs`                                                    missing-file             remark-validate-links
    37:391-37:442  warning  Link to unknown file: `https:/code.visualstudio.com`                                                          missing-file             remark-validate-links
    41:215-41:274  warning  Link to unknown file: `https:/www.conventionalcommits.org`                                                    missing-file             remark-validate-links
```

I tried setting `'repository': false,` as mentioned in the new README, but that didn't help.

### Your environment

*   **OS**: MacOS
*   **Packages**: 
    "remark-cli": "6.0.1",
    "remark-lint-heading-whitespace": "1.0.0",
    "remark-lint-no-url-trailing-slash": "3.0.1",
    "remark-preset-lint-consistent": "2.0.2",
    "remark-preset-lint-recommended": "3.0.2",
    "remark-validate-links": "remarkjs/remark-validate-links",
*   **Env**: node 8

Indeed, with node 8 I get

```
:remark-validate-links-bug kselden$ npm t

> remark-validate-links-bug@1.0.0 test /Users/kselden/code/remark-validate-links-bug
> remark -f README.md

### Test Filtering

In addition to the CLI filtering, covered [here](./README.md#filtering), you can single out a test in code using Mocha keywords. For any `describe`, `roles`, or `it`, you can add [`.only`](https://mochajs.org/#exclusive-tests) or [`.skip`](https://mochajs.org/#inclusive-tests). The `it`s also support `.allowFail` for known flakey tests via [mocha-helpers](https://github.com/kellyselden/mocha-helpers).
README.md
    3:43-3:72  warning  Link to unknown heading: `filtering`                                 missing-heading          remark-validate-links
  3:180-3:227  warning  Link to unknown file: `https:/mochajs.org`                           missing-file             remark-validate-links
  3:180-3:227  warning  Link to unknown heading in `https:/mochajs.org`: `exclusive-tests`   missing-heading-in-file  remark-validate-links
  3:231-3:278  warning  Link to unknown file: `https:/mochajs.org`                           missing-file             remark-validate-links
  3:231-3:278  warning  Link to unknown heading in `https:/mochajs.org`: `inclusive-tests`   missing-heading-in-file  remark-validate-links
  3:343-3:404  warning  Link to unknown file: `https:/github.com/kellyselden/mocha-helpers`  missing-file             remark-validate-links

⚠ 6 warnings
npm ERR! Test failed.  See above for more details.
```

and with node 12 I get what you got.

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,5 +1,5 @@
 language: node_js
 node_js:
-  - lts/dubnium
+  - lts/carbon
   - node
 after_success: bash <(curl -s https://codecov.io/bash)

--- a/lib/find/find.js
+++ b/lib/find/find.js
@@ -1,6 +1,7 @@
 'use strict'
 
 var path = require('path')
+var URL = require('url').URL // Node 8 support
 var vfile = require('to-vfile')
 var visit = require('unist-util-visit')
 var toString = require('mdast-util-to-string')

